app/src/main/java/com/codesoom/assignment/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,67 @@
+package com.codesoom.assignment.security;
+
+import com.codesoom.assignment.application.AuthenticationService;
+import com.codesoom.assignment.utils.JwtUtil;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+import org.springframework.util.StringUtils;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * Authorization 헤더를 통해 회원 인증을 수행하는 필터.
+ */
+public class JwtAuthenticationFilter extends BasicAuthenticationFilter {
+
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String BEARER = "Bearer ";
+
+    private final AuthenticationService authenticationService;
+    private final JwtUtil jwtUtil;
+
+    public JwtAuthenticationFilter(AuthenticationManager authenticationManager,
+                                   AuthenticationService authenticationService,
+                                   JwtUtil jwtUtil) {
+        super(authenticationManager);
+
+        this.authenticationService = authenticationService;
+        this.jwtUtil = jwtUtil;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain chain) throws IOException, ServletException {
+        resolveToken(request)
+                .map(accessToken -> {
+                    if (jwtUtil.validateToken(accessToken)) {
+                        return authenticationService.parseToken(accessToken);
+                    }
+                    return null;
+                })
+                .ifPresent(userId -> {
+                    SecurityContext context = SecurityContextHolder.getContext();
+                    UserAuthentication userAuthentication = new UserAuthentication(userId);
+                    context.setAuthentication(userAuthentication);
+                });
```

Comment:
아주 좋아요! 그런데 한번 쓰이고 버려지는 변수가 있군요. 객체 생성하는 메소드만으로 이해가 된다면 굳이 변수지정을 할 필요는 없어보여요! 아래 두가지 예시를 적어보았어요!

```suggestion
                .ifPresent(userId -> {
                    SecurityContextHolder.getContext()
                        .setAuthentication(new UserAuthentication(userId));
                });
```

혹은
```suggestion
                .map(UserAuthentication::new)
                .ifPresent(userAuthentication -> SecurityContextHolder
                               .getContext();
                               .setAuthentication(userAuthentication));
```

Reply:
아래 예시 참고해서 적용했습니다! 감사합니다